src/solvers/eigsolve.jl

```diff
@@ -0,0 +1,76 @@
+using Printf: @printf
+import ConstructionBase: setproperties
+
+@kwdef mutable struct EigsolveProblem{State,Operator}
+  state::State
+  operator::Operator
+  eigenvalue::Number = Inf
+end
+
+eigenvalue(E::EigsolveProblem) = E.eigenvalue
+ITensorNetworks.state(E::EigsolveProblem) = E.state
+operator(E::EigsolveProblem) = E.operator
+
+function updater(
+  E::EigsolveProblem,
```

Comment:
```suggestion
  problem::EigsolveProblem,
```
I think it would be helpful to stay consistent with the notation and always call the problem `problem`.